concert/processes/experiments.py

```diff
@@ -40,9 +40,9 @@ async def run(self):
 
             # prepare first iteration
             self._iteration = 0
-            await self._prepare_run(self._iteration)
 
             for iteration in range(await self.get_number_of_iterations()):
+                await self._prepare_run(self._iteration)
```

Comment:
I guess no point in having it twice.

Reply:
Oh wait, did you want `_prepare_next_run`? Anyway, this seems very coarse, I won't interfere until all your ideas are in properly.

Reply:
I think it should be correct now.